boards/shared/Inc/App/App_BoardEnum.h

```diff
@@ -2,11 +2,10 @@
 
 enum Board
 {
-    BMS,
+    GSM,
```

Comment:
GSM moved to enum value 0 such that the boards have the same order and values as Boardleds enum inside App_DriveState.c. GSM is not checked in DIM drive state, so it taking up position 0 where BMS used to be fits perfectly.

Snippet from DriveState:
```
typedef enum
{
    BMS_LED,
    DCM_LED,
    DIM_LED,
    FSM_LED,
    PDM_LED,
    NUM_BOARD_LEDS,
} BoardLeds;
```

Reply:
Hmm I don't see why this change is necessary. Why do the Board and BoardLed enums need the same order?

Reply:
It comes down to how the DIM checks for faults in each board. Because runs a for loop increment through the enum values listed above, for each value of i DIM calls App_SharedError_IsBoardInList, which uses the Board enum with the same i value. If they're not the same order it does not run correctly. The alternative fix would be to put BMS_LED at the bottom of BoardLeds, but the effect is the same.

Reply:
I think we should maintain the same order in both enums, that seems more clear. Also we should not delete BMS from this enum, since it is still a board in the car. That will preserve the same order for all relevant boards with LEDs, and since we only iterate up to NUM_BOARD_LEDs, GSM will be ignored.

Tbh this is not  ideal - we really should not have to worry about enum ordering for this to work. Maybe we refactor this, but that's outside the scope of this PR. For now, can you add a unit tests that just checks that both enums have the same ordering, i.e. check DCM == DCM_LED, etc., so if we accidentally change the ordering we'll know.

Reply:
I agree BMS should not be removed from this enum. The issue is that leaving BMS in the Board enum increases NUM_BOARDS by 1. NUM_BOARDS is used in many places to indicate to the shared error table how many boards to expect. I've adapted the behavior to always report the BMS as having no errors  inside of `App_SharedErrorTable_GetBoardsWithNoErrors` and adapted the tests to account for this, along with adding descriptive comments to explain the modifications. In the future when the all boards are removed from the table this kind of band-aid workaround won't be necessary